### PR TITLE
Add registrar and registrar's IANA ID to tracks event

### PR DESCRIPTION
**Pull request by delputnam**
This PR adds the registrar and registrar's IANA ID to the tracks event that is recorded for an inbound transfer domain when the "Continue" button is clicked on the pre-check page.

Dependent on D8762-code

To test this, start a new transfer, click "Continue" on the precheck screen.
Make sure that the tracks event shows the registrar and it's IANA ID in the event properties.

### CODE REVIEW

File: client/components/domains/transfer-domain-step/transfer-domain-precheck.jsx [hunk 1]
```diff
@@ -48,8 +48,9 @@ class TransferDomainPrecheck extends React.PureComponent {
 
 	onClick = () => {
 		const { domain, supportsPrivacy } = this.props;
+		const { registrar, registrarIanaId } = this.state;
```
Comment (hambai) [inline client/components/domains/transfer-domain-step/transfer-domain-precheck.jsx]: Those should be `losingRegistrar` and `losingRegistrarIanaId` :) It's not working right now

Comment (delputnam) [reply client/components/domains/transfer-domain-step/transfer-domain-precheck.jsx]: Ah...sorry. Fixed.

File: client/components/domains/transfer-domain-step/transfer-domain-precheck.jsx [hunk 2]
```diff
@@ -76,6 +77,8 @@ class TransferDomainPrecheck extends React.PureComponent {
 				privacy: result.privacy,
 				unlocked: result.unlocked,
 				loading: false,
+				losingRegistrar: result.registrar,
```
Comment (aidvu) [inline client/components/domains/transfer-domain-step/transfer-domain-precheck.jsx]: Probably smart to set defaults above. :)

File: client/components/domains/transfer-domain-step/transfer-domain-precheck.jsx [hunk 3]
```diff
@@ -354,8 +359,12 @@ class TransferDomainPrecheck extends React.PureComponent {
 const recordNextStep = ( domain_name, show_step ) =>
 	recordTracksEvent( 'calypso_transfer_domain_precheck_step_change', { domain_name, show_step } );
 
-const recordContinueButtonClick = domain_name =>
-	recordTracksEvent( 'calypso_transfer_domain_precheck_continue_click', { domain_name } );
+const recordContinueButtonClick = ( domain_name, losing_registrar, losing_registrar_iana_id ) =>
```
Comment (aidvu) [inline client/components/domains/transfer-domain-step/transfer-domain-precheck.jsx]: Oooh, I just saw we have these `snake_case` vars. We shouldn't. :)

Please use `camelCase` in Calypso always. Both events. Sorry. Unless we use it weirdly elsewhere, let's try to be consistent.

Comment (aidvu) [reply client/components/domains/transfer-domain-step/transfer-domain-precheck.jsx]: Edit: Disregard me... I see this everywhere. I don't know anymore what's right...

Comment (hambai) [reply client/components/domains/transfer-domain-step/transfer-domain-precheck.jsx]: Well - accodring to the tracks naming convention properties have to be in `snake_case` (see https://fieldguide.automattic.com/tracks/tracks-for-developers-product-teams/tracks-naming-conventions/ ). However I agree we should not have `snake_case` variables so we'd better construct the event properties as `{ domain_name: domainName, losing_registrar: losingRegistrar }`

Comment (hambai) [reply client/components/domains/transfer-domain-step/transfer-domain-precheck.jsx]: Let's do this in a separate PR? I'll go through all the tracks events for incoming transfers and fix it

**Pull request merged**